shared/java/com/google/idea/blaze/common/Label.java

```diff
@@ -57,17 +56,22 @@ public static Label of(String label) {
   }
 
   public static Label fromWorkspacePackageAndName(String workspace, Path packagePath, Path name) {
-    final var systemIndependentPackagePath = StreamSupport.stream(packagePath.spliterator(), false)
-        .map(Path::toString)
-        .collect(Collectors.joining("/"));
-
+    String packageWithForwardSlashes = withForwardSlashes(packagePath);
+    String nameWithForwardSlashes = withForwardSlashes(name);
     if (workspace.isEmpty()) {
-      return Label.of(String.format("//%s:%s", systemIndependentPackagePath, name));
+      return Label.of(String.format("//%s:%s", packageWithForwardSlashes, nameWithForwardSlashes));
     } else {
-      return Label.of(String.format("@@%s//%s:%s", workspace, systemIndependentPackagePath, name));
+      return Label.of(String.format("@@%s//%s:%s", workspace, packageWithForwardSlashes, nameWithForwardSlashes));
     }
   }
 
+  private static String withForwardSlashes(Path p) {
```

Comment:
`com.intellij.openapi.util.io.FileUtilRt#toSystemIndependentName`

Reply:
It turns out not so easy to refactor. This package is pretty low in the dependency hierarchy, and it doesn't depend on the IntelliJ API.